sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -484,6 +488,59 @@ case class GpuCast(
     }
   }
 
+  private def castStructToString(input: GpuColumnVector): ColumnVector = {
+    var separatorColumn: ColumnVector = null
+    var spaceColumn: ColumnVector = null
+    val columns: ArrayBuffer[ColumnVector] = new ArrayBuffer[ColumnVector]()
+    val coreColumns: ArrayBuffer[ColumnVector] = new ArrayBuffer[ColumnVector]()
+
+    try {
+      withResource(GpuScalar.from(",", StringType)) { separatorScalar =>
+        separatorColumn = ColumnVector.fromScalar(separatorScalar, input.getRowCount().toInt)
+      }
+      withResource(GpuScalar.from(" ", StringType)) { separatorScalar =>
+        spaceColumn = ColumnVector.fromScalar(separatorScalar, input.getRowCount().toInt)
+      }
+      withResource(GpuScalar.from("[", StringType)) { bracketScalar =>
+        columns += ColumnVector.fromScalar(bracketScalar, input.getRowCount().toInt)
+      }
+      withResource(input.getBase().getChildColumnView(0)) { childView =>
+        columns += childView.asStrings()
```

Comment:
Why are we not recursively having the sub-columns cast to strings? There is a lot of formatting that this is not going to properly cover.

Reply:
This was pretty nasty to implement because getting the struct column's child columns only returns `ColumnView`. Recursively calling `doColumnar` required copying each column to create a `ColumnVector`. These copies are unnecessary as far as I can tell, may be worth a followup PR.